### PR TITLE
ISPN-6673 Rolling Upgrades for Kubernetes/OpenShift

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/ISPN-6673

Still missing:
- Kubernetes Rolling Update and persisting the configuration

### CODE REVIEW

File: documentation/src/main/asciidoc/user_guide/cloud.adoc [hunk 1]
```diff
@@ -62,3 +62,23 @@ Finally you can configure your JGroups instances to use a shared storage to exch
...
+
+.Key differences when upgrading using OpenShift/Kubernetes are:
+* When forming a new cluster, make sure you use labels. Labels can be used by link:$$https://github.com/jgroups-extras/jgroups-kubernetes$$[Kubernetes PING Protocol] (see `OPENSHIFT_KUBE_PING_LABELS` environment variable) and they allow controlling which nodes are assigned to which clusters.
+* Depending on configuration, it is a good practice to use link:$$https://docs.openshift.org/latest/architecture/core_concepts/routes.html$$[OpenShift Routes] or link:$$http://kubernetes.io/docs/user-guide/ingress$$[Kubernetes Ingress API] to expose services to the clients. During the upgrade the Route (or Ingress) used by the clients can be altered to point to the new cluster.
+* Invoking CLI commands can be done by using Kubernetes (`kubectl exec`) or OpenShift clients (`oc exec`). Here is an example: `oc exec <POD_NAME> -- '/opt/jboss/infinispan-server/bin/ispn-cli.sh' '-c' '--controller=$(hostname -i):9990' '/subsystem=datagrid-infinispan/cache-container=clustered/distributed-cache=default:disconnect-source(migrator-name=hotrod)'`
```
Comment (danberindei) [inline documentation/src/main/asciidoc/user_guide/cloud.adoc]: Hmmm, `hostname -i` gives `fe80::xxxx:xxxx:xxxx:xxxx%wlp4s0 10.x.x.x 192.168.x.x` on my machine. Are OpenShift hosts guaranteed to have just 1 IPv4 address?

Comment (slaskawi) [inline documentation/src/main/asciidoc/user_guide/cloud.adoc]: I believe so. Here's a nice snippet from Docker [Advanced Networking guide](https://docs.docker.com/v1.5/articles/networking/):

> hen Docker starts, it creates a virtual interface named docker0 on the host machine. It randomly chooses an address and subnet from the private range defined by RFC 1918 that are not in use on the host machine, and assigns it to docker0. Docker made the choice 172.17.42.1/16 when I started it a few minutes ago, for example — a 16-bit netmask providing 65,534 addresses for the host machine and its containers

**Comment by galderz** [comment]
Needs rebasing

**Comment by galderz** [comment]
Needs rebasing. Reopen when fixed.

**Pull request closed without merging**